### PR TITLE
Add DVCLive snippets

### PR DESCRIPTION
**Pull request by mattseddon**
This PR adds DVCLive snippets for Catalyst, Fast.ai, Hugging Face, Keras, LightGBM, Optuna, Pytorch Lightning & XGBoost. The other examples seem too involved to be useful as snippets (the omitted frameworks are MMCV, Pytorch & TensorFlow)

I have also added a note to the `<DVCLiveExamples />` component which mentions the existence of these snippets.

### Demo

https://github.com/iterative/vscode-dvc/assets/37993418/c0209fd5-2160-442d-872c-9bef69520dc4

cc @daavoo (feedback appreciated)

### CODE REVIEW

File: extension/snippets/dvclive.code-snippets [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      "    callbacks=[DVCLiveCallback()])"
+    ],
+    "description": "DVCLive Catalyst callback"
+  },
+  "DVCLive Fast.ai": {
+    "prefix": "dvclive-fastai",
+    "body": [
+      "from dvclive.fastai import DVCLiveCallback",
+      "",
+      "learn = tabular_learner(${1:data_loader}, metrics=${2:accuracy})",
```
Comment (daavoo) [inline extension/snippets/dvclive.code-snippets:18]: Let's not include it, as it is specific to tabular data, and there is no easy way to generalize 

```suggestion
```

**Review by daavoo — APPROVED** [review]
@dberenbaum I don't know if it would make a difference but maybe we should invest in having some consistent, minimally usable snippets across docs/products.

Feels like we differ too much from Lightning being almost copy-pasteable and others like Keras being too far from usable

**Review by julieg18 — APPROVED** [review]
Great work!

**Comment by dberenbaum** [comment]
Thanks @daavoo! Let's make it part of https://iterative-squad.monday.com/boards/2546688237

**Comment by mattseddon** [comment]
Contributions are welcome when the snippets get updated 😅 🙏🏻.

**Pull request merged**